### PR TITLE
add sequelize to 3rd party tools incompatibility

### PR DESCRIPTION
**Pull request by shczhen**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/11654
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by shczhen** [comment]
Target ZH doc: https://github.com/pingcap/docs-cn/pull/11654

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 1]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+- `PROCEDURE` is not supported;
+- The [isolation level](/system-variables.md#transaction_isolation) `READ-UNCOMMITTED` and `SERIALIZABLE` are not supported;
+- Modification of a column's `AUTO_INCREMENT` property is not allowed by default;
+- `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported;
+
+### Modification of Integer Primary Key is not supported
+
+**Description**
+
+Modification of Integer Primary Key is not supported. TiDB uses primary key as an index for data organization if the primary key is integer type. Refer to [Issue #18090](https://github.com/pingcap/tidb/issues/18090) and [Clustered Indexes](/clustered-indexes.md) for more details.
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:180]:
```suggestion
Modification of integer primary key is not supported. TiDB uses primary key as an index for data organization if the primary key is integer type. Refer to [Issue #18090](https://github.com/pingcap/tidb/issues/18090) and [Clustered Indexes](/clustered-indexes.md) for more details.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 2]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+### The isolation level `READ-UNCOMMITTED` and `SERIALIZABLE` are not supported
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:182]:
```suggestion
### The `READ-UNCOMMITTED` and `SERIALIZABLE` isolation levels are not supported
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 3]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+**Description**
+
+TiDB does not support isolation level `READ-UNCOMMITTED` and `SERIALIZABLE`. TiDB will throw an error if the isolation level is set to `READ-UNCOMMITTED` or `SERIALIZABLE`.
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:186]:
```suggestion
TiDB does not support the `READ-UNCOMMITTED` and `SERIALIZABLE` isolation levels. If the isolation level is set to `READ-UNCOMMITTED` or `SERIALIZABLE`, TiDB throws an error.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 4]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+**Way to avoid**
+
+Use the `REPEATABLE-READ` or the `READ-COMMITTED` isolation level only which TiDB support.
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:190]:
```suggestion
Use only the isolation level that TiDB supports: `REPEATABLE-READ` or `READ-COMMITTED`.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 5]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+You can set [`tidb_skip_isolation_level_check`](/system-variables.md#tidb_skip_isolation_level_check) to `1` if you want to be compatible with but do not depend on other applications that set the `SERIALIZABLE` isolation level. Then TiDB will ignore the unsupported isolation level error.
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:192]:
```suggestion
If you want TiDB to be compatible with other applications that set the `SERIALIZABLE` isolation level but not depend on `SERIALIZABLE`, you can set [`tidb_skip_isolation_level_check`](/system-variables.md#tidb_skip_isolation_level_check) to `1`. In such case, TiDB ignores the unsupported isolation level error.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 6]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+### Modification of a column's `AUTO_INCREMENT` property is not allowed by default
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:194]:
```suggestion
### Modification of a column's `AUTO_INCREMENT` attribute is not allowed by default
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 7]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+Adding or removing the `AUTO_INCREMENT` attribute of a column via `ALTER TABLE MODIFY` or `ALTER TABLE CHANGE` command is not allowed by default.
+
+**Way to avoid**
+
+Refer to the [restrictions of `AUTO_INCREMENT`](/auto-increment.md#restrictions).
+
+Set `@@tidb_allow_remove_auto_inc` to `true` to allow removal of the `AUTO_INCREMENT` attribute.
+
+### `FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:206]:
```suggestion
### `FULLTEXT`, `HASH`, and `SPATIAL` indexes are not supported
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 8]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+**Description**
+
+`FULLTEXT`, `HASH` and `SPATIAL` indexes are not supported.
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:210]:
```suggestion
`FULLTEXT`, `HASH`, and `SPATIAL` indexes are not supported.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 9]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+Unsupported features are:
+
+- `Foreign Key` constraints (including many-to-many relationships) are not supported;
+- [`GEOMETRY`](https://github.com/pingcap/tidb/issues/6347) is not supported;
+- Modification of integer primary key is not supported;
+- `PROCEDURE` is not supported;
+- The `READ-UNCOMMITTED` and `SERIALIZABLE` [isolation levels](/system-variables.md#transaction_isolation) are not supported;
+- Modification of a column's `AUTO_INCREMENT` attribute is not allowed by default;
+- `FULLTEXT`, `HASH`, and `SPATIAL` indexes are not supported;
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:174]:
```suggestion
- `Foreign Key` constraints (including many-to-many relationships) are not supported.
- [`GEOMETRY`](https://github.com/pingcap/tidb/issues/6347) is not supported.
- Modification of integer primary key is not supported.
- `PROCEDURE` is not supported.
- The `READ-UNCOMMITTED` and `SERIALIZABLE` [isolation levels](/system-variables.md#transaction_isolation) are not supported.
- Modification of a column's `AUTO_INCREMENT` attribute is not allowed by default.
- `FULLTEXT`, `HASH`, and `SPATIAL` indexes are not supported.
```

**Review by ran-huang — APPROVED** [review]

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 10]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+## Compatibility with Sequelize
+
+The compatibility information described in this section is based on [Sequelize v6.21.4](https://www.npmjs.com/package/sequelize/v/6.21.4).
+
+According to the test results, TiDB supports most of the Sequelize features ([using `MySQL` as the dialect](https://sequelize.org/docs/v6/other-topics/dialect-specific-things/#mysql)).
+
+Unsupported features are:
+
+- `Foreign Key` constraints (including many-to-many relationships) are not supported.
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:168]:
```suggestion
- Foreign key constraints (including many-to-many relationships) are not supported.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 11]
```diff
@@ -156,3 +156,55 @@ TiDB fixes it in the following ways:
...
+
+**Description**
+
+Adding or removing the `AUTO_INCREMENT` attribute of a column via `ALTER TABLE MODIFY` or `ALTER TABLE CHANGE` command is not allowed by default.
+
+**Way to avoid**
+
+Refer to the [restrictions of `AUTO_INCREMENT`](/auto-increment.md#restrictions).
+
+Set `@@tidb_allow_remove_auto_inc` to `true` to allow removal of the `AUTO_INCREMENT` attribute.
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:204]:
```suggestion
To allow the removal of the `AUTO_INCREMENT` attribute, set `@@tidb_allow_remove_auto_inc` to `true`.
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**